Makefile

```diff
@@ -31,97 +35,109 @@ endif
 CHMOD		:= chmod
 SED		:= sed
 TAR		:= tar
-LINK		= $(CC) $(OBJS-$@) -o $@ $(LDFLAGS) $(LDFLAGS-$@) $(LIBS-$@)
+COMPILE		= $(CC) $(CPPFLAGS) $(CFLAGS) $(CFLAGS-$@) -c $^
+LINK		= $(CC) -o $@ $^ $(LDFLAGS) $(LDFLAGS-$@) $(LIBS-$@)
 
 SED_REPLACE	:= -e 's:@VERSION@:$(FULL_VERSION):g' \
 			-e 's:@prefix@:$(prefix):g' \
 			-e 's:@sysconfdir@:$(sysconfdir):g' \
 			-e 's:@datadir@:$(datadir):g' \
 
+ABUILD_DEPS	+= openssl-dev
 SSL_CFLAGS	?= $(shell pkg-config --cflags openssl)
 SSL_LDFLAGS	?= $(shell pkg-config --cflags openssl)
 SSL_LIBS	?= $(shell pkg-config --libs openssl)
-ZLIB_LIBS	?= $(shell pkg-config --libs zlib)
-
-OBJS-abuild-tar  = abuild-tar.o
-CFLAGS-abuild-tar.o = $(SSL_CFLAGS)
-LDFLAGS-abuild-tar = $(SSL_LDFLAGS)
-LIBS-abuild-tar = $(SSL_LIBS)
-LIBS-abuild-tar.static = $(LIBS-abuild-tar)
-
-OBJS-abuild-gzsplit = abuild-gzsplit.o
-LDFLAGS-abuild-gzsplit = $(ZLIB_LIBS)
-
-OBJS-abuild-sudo = abuild-sudo.o
-OBJS-abuild-fetch = abuild-fetch.o
-
-.SUFFIXES:	.sh.in .in
-%.sh: %.sh.in
-	${SED} ${SED_REPLACE} ${SED_EXTRA} $< > $@
-	${CHMOD} +x $@
 
-%: %.in
-	${SED} ${SED_REPLACE} ${SED_EXTRA} $< > $@
-	${CHMOD} +x $@
+ABUILD_EXT	:= abuild-tar
+CFLAGS-$(ABUILD_EXT).o = $(SSL_CFLAGS)
+LDFLAGS-$(ABUILD_EXT) = $(SSL_LDFLAGS)
+LIBS-$(ABUILD_EXT) = $(SSL_LIBS)
+ABUILD_EXT	:=
 
-P=$(PACKAGE)-$(VERSION)
+ABUILD_DEPS	+= zlib-dev
+ZLIB_CFLAGS	?= $(shell pkg-config --cflags zlib)
+ZLIB_LDFLAGS	?= $(shell pkg-config --cflags zlib)
+ZLIB_LIBS	?= $(shell pkg-config --libs zlib)
 
-all:	$(USR_BIN_FILES) functions.sh
+ABUILD_EXT	:= abuild-gzsplit
+CFLAGS-$(ABUILD_EXT).o = $(ZLIB_CFLAGS)
+LDFLAGS-$(ABUILD_EXT) = $(ZLIB_LDFLAGS)
+LIBS-$(ABUILD_EXT) = $(ZLIB_LIBS)
+ABUILD_EXT	:=
 
-clean:
-	@rm -f $(USR_BIN_FILES) functions.sh
+all: templates bins.static bins
 
-%.o: %.c
-	$(CC) $(CPPFLAGS) $(CFLAGS) $(CFLAGS-$@) -o $@ -c $<
+bins: $(BINS) undeps
 
-abuild-sudo: abuild-sudo.o
-	$(LINK)
+bins.static: $(addsuffix .static,$(BINS))
 
-abuild-tar: abuild-tar.o
-	$(LINK)
+clean: mostlyclean | undeps
+	@rm -f functions.sh $(SCRIPTS)
+	@rm -f $(BINS) $(addsuffix .static,$(BINS))
+	@rm -f $(addsuffix .o,$(BINS))
 
-abuild-fetch: abuild-fetch.o
-	$(LINK)
-
-abuild-gzsplit: abuild-gzsplit.o
-	$(LINK)
-
-abuild-tar.static: abuild-tar.o
-	$(CC) -static $(CPPFLAGS) $(CFLAGS) $(CFLAGS-$@) $^ -o $@ $(LIBS-$@)
+deps:
+	@abuild-apk add --virtual .$(PACKAGE)-makedepends $(ABUILD_DEPS)
```

Comment:
The Makefile should not install/uninstall build time dependencies.

Reply:
Please explain why you think the Makefile should not contain the steps required to make the targets it defines.

Reply:
Because it breaks portability. You will no longer be able to build abuild on other distros or macOS. It is not the Makefiles' responsibility to pull in dependencies.